src/qibolab/instruments/erasynth.py

```diff
@@ -22,37 +22,28 @@ class ERA(LocalOscillator):
     def __init__(self, name, address, ethernet=True, reference_clock_source="internal"):
         super().__init__(name, address)
         self.device: ERASynthPlusPlus = None
-        self._power: int = None
-        self._frequency: int = None
+        self.settings = LocalOscillatorSettings()
         self.ethernet = ethernet
         self._device_parameters = {}
         self.reference_clock_source = reference_clock_source
 
     @property
     def frequency(self):
-        if self.is_connected:
-            if self.ethernet:
-                return int(self._get("frequency"))
-            return self.device.get("frequency")
-        return self._frequency
+        return self.settings.frequency
```

Comment:
Here you're getting rid of the connection check.

It is not even restored on the other side. Is there any reason for that?

Reply:
I see, it was not shown in the "Files changed" tab, and I forgot. Sorry.